### DISCUSSION TITLE
Update NextAuth.js Schema in Vercel "Fullstack App Guide"

### DISCUSSION THREAD

**Discussion by ndom91** [prisma/prisma#12365] [Q&A] [answered]
Hi Team,

Big fan of Prisma! 

Just wanted to note that the Prisma schema provided for [NextAuth.js](https://next-auth.js.org) in this [Vercel guide](https://vercel.com/guides/nextjs-prisma-postgres#step-5.-set-up-github-authentication-with-nextauth) (Written by @nikolasburk) is unfortunately wrong. 

We only found out as someone came to the [issues](https://github.com/nextauthjs/next-auth/issues/3831#issuecomment-1066031131) writing about not being able to query their Sessions correctly after having setup NextAuth.js via that guide. 

Anyway, the specific issue is that the `id` of the `Session` model should be a `String`, not an `Int`.

The working schema is as follows:

```prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
  shadowDatabaseUrl = env("SHADOW_DATABASE_URL") // Only needed when using a cloud provider that doesn't support the creation of new databases, like Heroku. Learn more: https://pris.ly/migrate-shadow
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["referentialActions"] // You won't need this in Prisma 3.X or higher.
}

model Account {
  id                 String  @id @default(cuid())
  userId             String
  type               String
  provider           String
  providerAccountId  String
  refresh_token      String?  @db.Text
  access_token       String?  @db.Text
  expires_at         Int?
  token_type         String?
  scope              String?
  id_token           String?  @db.Text
  session_state      String?

  user User @relation(fields: [userId], references: [id], onDelete: Cascade)

  @@unique([provider, providerAccountId])
}

model Session {
  id           String   @id @default(cuid())
  sessionToken String   @unique
  userId       String
  expires      DateTime
  user         User     @relation(fields: [userId], references: [id], onDelete: Cascade)
}

model User {
  id            String    @id @default(cuid())
  name          String?
  email         String?   @unique
  emailVerified DateTime?
  image         String?
  accounts      Account[]
  sessions      Session[]
}

model VerificationToken {
  identifier String
  token      String   @unique
  expires    DateTime

  @@unique([identifier, token])
}
```

For more details, check out [this docs page](https://next-auth.js.org/adapters/prisma#create-the-prisma-schema).

If yall have some time, would be awesome to get that updated :grin:

**Comment by nurul3101** [comment, marked as answer]
Hey @ndom91 👋 

Thank you for bringing this to notice. I did go through the article and can see it needs to be updated, there are certain dead links as well in that article that can be fixed.

I'll ask if our team can directly update it or if we need to request access from the vercel team.

**Comment by bhajneet** [comment]
@ndom91 Thank you for this! I was very confused how we were getting User but no Session/Account. Looking through debug logs I was probably going to get here eventually, but you helped speed up that process.

**Comment by yagudaev** [comment]
I want to highlight the `shadowDatabaseUrl` and `directDatabaseUrl` usage there is wrong. 

The article uses `prisma db push`, which works for quick prototyping. But when deployed you will need to use `prisma migrate dev`. 

The shadow database defined implies it is the same database, just with no connection pooling. However the purpose is for comparing the state of the database using a diff to figure out migrations.

Specifying the same DB means Prisma will try to delete the shadow database and run all the migrations again on a single database. This causes a lot of weird behaviour and breaks the workflow and metal model of Prisma.

Here is the correct documentation: 
1. Shadow Database on cloud providers: https://www.prisma.io/docs/concepts/components/prisma-migrate/shadow-database#cloud-hosted-shadow-databases-must-be-created-manually
2. External Connection poolers: https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#external-connection-poolers

**Reply by ndom91** [reply]
Thanks for pointing these out. 

So other than the command push/migrate change, i.e. note to use one in dev and one in prod, what else would you recommend changing specifically?

**Reply by janpio** [reply]
The `shadowDatabaseUrl` that was in the guide got removed because of exactly the points that @yagudaev raised. The guide (and other documentation) should now be up to date.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
